### PR TITLE
Allow for CFONB files without linebreaks

### PR DESCRIPTION
**Pull request by tschanzt**

### CODE REVIEW

**Review by jcoux — CHANGES_REQUESTED** [review]
Can you add comment in code and explain your modification in PR description?
Also example of required content files can help to understand.

**Comment by tschanzt** [comment]
@jcoux i added a comment

**Review by jcoux — APPROVED** [review]
Thanks for explanations.

File: account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py [hunk 1]
```diff
@@ -57,21 +57,21 @@ def _parse_file(self, data_file):
...
                 data_file)
         transactions = []
-        if not data_file.splitlines():
+        # The CFONB spec says you should only have digits, capital letters
+        # and * - . /
+        # But many banks don't respect that and use regular letters for exemple
+        # And I found one (LCL) that even uses caracters with accents
+        # So the best method is probably to decode with latin1
+        data_file_decoded = data_file.decode('latin1')
+        if not self.split_lines(data_file_decoded):
```
Comment (i-vyshnevska) [inline account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py]:
```suggestion
        lines = self.split_lines(data_file_decoded)
        if not lines:
        
```

File: account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py [hunk 2]
```diff
@@ -57,21 +57,21 @@ def _parse_file(self, data_file):
...
         start_balance = end_balance = False
         vals_line = False
-        # The CFONB spec says you should only have digits, capital letters
-        # and * - . /
-        # But many banks don't respect that and use regular letters for exemple
-        # And I found one (LCL) that even uses caracters with accents
-        # So the best method is probably to decode with latin1
-        data_file_decoded = data_file.decode('latin1')
-        for line in data_file_decoded.splitlines():
+        for line in self.split_lines(data_file_decoded):
```
Comment (i-vyshnevska) [inline account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py]:
```suggestion
        for line in lines:
```

File: account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py [hunk 3]
```diff
@@ -158,5 +158,32 @@ def _parse_file(self, data_file):
...
+        data_file_without_linebreaks = data_file.replace(
+            '\n', ''
+        ).replace('\r', '')
+
+        # check length of file
+        max_len = len(data_file_without_linebreaks)
+        lines = []
+
+        # make sure the length is a multiple of 120 otherwise it isn't valid
+        if max_len % 120:
```
Comment (i-vyshnevska) [inline account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py]: 120 can be a constant in this case

File: account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py [hunk 4]
```diff
@@ -57,21 +57,21 @@ def _parse_file(self, data_file):
...
         transactions = []
-        if not data_file.splitlines():
+        # The CFONB spec says you should only have digits, capital letters
+        # and * - . /
+        # But many banks don't respect that and use regular letters for exemple
+        # And I found one (LCL) that even uses caracters with accents
+        # So the best method is probably to decode with latin1
+        data_file_decoded = data_file.decode('latin1')
+        if not self.split_lines(data_file_decoded):
             raise UserError(_('The file is empty.'))
```
Comment (i-vyshnevska) [inline account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py]:
```suggestion
```
can be also raised in split_lines

File: account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py (hunk 3 above)
Comment (sebalix) [inline account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py]: So we are enforcing the CFONB spec here, with maybe generating regressions in current module installations?
Should we fallback on the previous behavior (so splitting lines whatever the size of the lines are)? It is just a question, not blocking. If every banks respect the rule of 120 characters we are fine.

Comment (tschanzt) [reply account_bank_statement_import_fr_cfonb/wizard/account_bank_statement_import.py]: Well that is what i got from @CharlineDumontet, i would imagine the 120 chars are more important if there is banks sending the file around without newline. Sadly i can not read the spec as my french is not good enough to navigate the cfonb site to find it (i suspect there might not be a english version of the spec available as well)

**Review by sebalix — APPROVED** [review]

**Pull request merged**